nmstatectl/nmstatectl.py

```diff
@@ -50,12 +54,20 @@ def setup_subcommand_set(subparsers):
 
 
 def show(args):
-    print(json.dumps(netinfo.show(), indent=4, sort_keys=True,
-                     separators=(',', ': ')))
+    state = netinfo.show()
+    if args.yaml:
+        print(yaml.dump(state, default_flow_style=False))
+    else:
+        print(json.dumps(state, indent=4, sort_keys=True,
+                         separators=(',', ': ')))
 
 
 def apply(args):
     with open(args.file) as f:
-        state = json.load(f)
-        netapplier.apply(state)
+        statedata = f.read()
+    try:
```

Comment:
I find this problematic, because it shadows a bad json file.
So either the "set" should also have a "--yaml" flag, or attempt to guess the file type from the file extension.

Reply:
I changed it to try only of each types depending on the first character.

Reply:
Nice.